layers/org/packages.el

```diff
@@ -110,6 +110,12 @@
             ;; `helm-org-headings-max-depth'.
             org-imenu-depth 8)
 
+      (with-eval-after-load 'org
+        (when (boundp 'org-plantuml-jar-path)
+          (org-babel-do-load-languages
+           'org-babel-load-languages
+           '((plantuml . t)))))
+
```

Comment:
Org layer should not know about plantuml. Do it the other way around:
add `org` to the `plantuml` layer package list and move this configuration to `plantuml/post-init-org` function. Since the configuration lies in the `plantuml` layer you don't need to check for `org-plantum-jar-path`, and `plantuml/post-init-org` is executed only if the `org` layer is used.
You can either keep the `with-eval-after-load` or use a `use-package` hook with `spacemacs|use-package-add-hook` to inject your configuration in the `:config` block for instance.